grails-app/domain/org/pih/warehouse/core/Location.groovy

```diff
@@ -66,7 +66,13 @@ class Location implements Comparable<Location>, java.io.Serializable {
     static mappedBy = [locations: "parentLocation"]
 
     static constraints = {
-        name(nullable: false, blank: false, maxSize: 255, unique: 'parentLocation')
+        name(nullable: false, blank: false, maxSize: 255, unique: 'parentLocation', validator: { name, obj ->
+            Boolean isLocationDuplicated = obj.parentLocation ?
+                    Location.findAllByNameAndParentLocation(name, obj.parentLocation).size() > 1 :
+                    Location.findAllByNameAndParentLocationIsNull(name).size() > 1
+
+            return isLocationDuplicated ? ['duplication.error'] : true
```

Comment:
Let's use the same error code as the standard unique constraint error. 
```
default.not.unique.message=Property [{0}] of class [{1}] with value [{2}] must be unique
```
![image](https://user-images.githubusercontent.com/1162935/231356568-9708a756-c607-4376-a257-d785796a3c5b.png)